### PR TITLE
Release process automatization

### PR DESCRIPTION
**Pull request by ccraluca**
Each new version tag will trigger "PushToGithubRelease" job to run on Azure Pipelines. The result of this job is a draft release on Github.

Draft release will contain:
    - buildartifacts
    - changelog
    - release title.

CI/travis/archive_artifacts.sh script was added in order to archive buildartifacts that will be pushed as assets in draft release.

### CODE REVIEW

**Comment by ccraluca** [comment]
V1:
- fix codacy errors

File: azure-pipelines.yml [hunk 1]
```diff
@@ -175,5 +181,28 @@ stages:
       - bash: chmod 600 $(key.secureFilePath) ; scp -2 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o HostKeyAlgorithms=+ssh-dss -vv -i $(key.secureFilePath) -r /home/vsts/work/1/a/* $MAPPED_VAR
         env:
           MAPPED_VAR: $(SERVER_ADDRESS)
-        condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/master'))
         displayName: "Push artifacts to SW Downloads"
+  - job: PushToGithubRelease
+    condition: startsWith(variables['Build.SourceBranch'], 'refs/tags/v')
```
Comment (tfcollins) [inline azure-pipelines.yml:186]: Can we add to this condition that the build succeeded like PushToSWDownloads

File: CI/travis/archive_artifacts.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	local linux_dist='CentOS-7-x86_64 CentOS-8-x86_64 Ubuntu-16.04-x86_64 Ubuntu-18.04-x86_64 Ubuntu-20.04-x86_64 Debian-Buster-ARM Debian-Buster-ARM64'
+	cd "${SOURCE_DIRECTORY}"
+	for distribution in $linux_dist; do
+		tar -zcvf Linux-"${distribution}".tar.gz Linux-"${distribution}"
+		rm -r Linux-"${distribution}"
+	done
+}
+
+archive_macOS() {
+        local macOS_dist='10.14 10.15'
```
Comment (nunojsa) [inline CI/travis/archive_artifacts.sh:13]: new line

File: CI/travis/archive_artifacts.sh [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        cd "${SOURCE_DIRECTORY}"
+        for distribution in $macOS_dist; do
+                tar -zcvf macOS-"${distribution}".tar.gz macOS-"${distribution}"
+		rm -r macOS-"${distribution}"
```
Comment (nunojsa) [inline CI/travis/archive_artifacts.sh:17]: indentation...

File: CI/travis/archive_artifacts.sh [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        done
+}
+
+archive_windows() {
+        local windows_dist='Win32 x64'
```
Comment (nunojsa) [inline CI/travis/archive_artifacts.sh:22]: ditto

File: CI/travis/archive_artifacts.sh [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        cd "${SOURCE_DIRECTORY}"
+        for distribution in $windows_dist; do
+                zip -r Windows-VS-16-2019-"${distribution}".zip Windows-VS-16-2019-"${distribution}"
+		rm -r Windows-VS-16-2019-"${distribution}"
```
Comment (nunojsa) [inline CI/travis/archive_artifacts.sh:26]: ditto

**Comment by ccraluca** [comment]
V2:
- add succeeded condition
- add new lines between the variable declaration and the function implementation
- break long line

**Review by nunojsa — APPROVED** [review]

**Review by tfcollins — APPROVED** [review]

**Pull request merged**